hack/update-codegen.sh

```diff
@@ -1,124 +1,48 @@
-echo "Generating client,lister,informer for kubeflow.org/v1 and kubeflow.org/v2alpha1"
-${CODEGEN_PKG}/generate-groups.sh "client,lister,informer" \
-    github.com/kubeflow/training-operator/pkg/client github.com/kubeflow/training-operator/pkg/apis \
-    kubeflow.org:v1,v2alpha1 \
-    --output-base "${TEMP_DIR}" \
-    --go-header-file hack/boilerplate/boilerplate.go.txt
-
-# Notice: The code in code-generator does not generate defaulter by default.
-# We need to build binary from vendor cmd folder.
-#echo "Building defaulter-gen"
-#go build -o defaulter-gen ${CODEGEN_PKG}/cmd/defaulter-gen
-
-# $(go env GOPATH)/bin/defaulter-gen is automatically built from ${CODEGEN_PKG}/generate-groups.sh
-echo "Generating defaulters for kubeflow.org/v1"
-$(go env GOPATH)/bin/defaulter-gen --input-dirs github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1 \
-    -O zz_generated.defaults \
-    --output-package github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1 \
-    --go-header-file hack/boilerplate/boilerplate.go.txt "$@" \
-    --output-base "${TEMP_DIR}"
-
-cd - >/dev/null
-
-# Notice: The code in kube-openapi does not generate defaulter by default.
-# We need to build binary from pkg cmd folder.
-echo "Building openapi-gen"
-go build -o openapi-gen ${OPENAPI_PKG}/cmd/openapi-gen
-
-echo "Generating OpenAPI specification for kubeflow.org/v1"
-./openapi-gen --input-dirs github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1 \
-    --report-filename=hack/violation_exception_v1.list \
-    --output-package github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1 \
-    --go-header-file hack/boilerplate/boilerplate.go.txt "$@" \
-    --output-base "${TEMP_DIR}"
-
-echo "Generating OpenAPI specification for kubeflow.org/v2alpha1"
-./openapi-gen --input-dirs github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v2alpha1 \
-    --report-filename=hack/violation_exception_v2alpha1.list \
-    --output-package github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v2alpha1 \
-    --go-header-file hack/boilerplate/boilerplate.go.txt "$@" \
-    --output-base "${TEMP_DIR}"
-
-echo "Generate OpenAPI Swagger for kubeflow.org/v2alpha1"
-go run hack/swagger-v2/main.go >api.v2/openapi-spec/swagger.json
-
-cd - >/dev/null
-
-# Copy everything back.
-cp -a "${TEMP_DIR}/${ROOT_PKG}/." "${SCRIPT_ROOT}/"
-# Clean up binaries we build for update codegen
-rm ./openapi-gen
+GO_CMD=${1:-go}
+CURRENT_DIR=$(dirname "${BASH_SOURCE[0]}")
+TRAINING_OPERATOR_ROOT=$(realpath "${CURRENT_DIR}/..")
+TRAINING_OPERATOR_PKG="github.com/kubeflow/training-operator"
+CODEGEN_PKG=$(go list -m -mod=readonly -f "{{.Dir}}" k8s.io/code-generator)
+
+cd "$CURRENT_DIR/.."
+
+# shellcheck source=/dev/null
+source "${CODEGEN_PKG}/kube_codegen.sh"
+
+# Generating conversion and defaults functions
+kube::codegen::gen_helpers \
+  --boilerplate "${TRAINING_OPERATOR_ROOT}/hack/boilerplate/boilerplate.go.txt" \
+  "${TRAINING_OPERATOR_ROOT}/pkg/apis"
```

Comment:
@astefanutti @tenzen-y Is this a new recommended way to use kube codegen ?

Reply:
Yes, the previous command has been completely removed.

Reply:
Right, the "old" way has recently been EOL'ed: kubernetes/code-generator@2e5be312636d3c77fb42ba37cedd4707968608a0.